homeassistant/components/co2signal/sensor.py

```diff
@@ -48,59 +78,47 @@ async def async_setup_platform(hass, config, add_entities, discovery_info=None):
 
 async def async_setup_entry(hass, entry, async_add_entities):
     """Set up the CO2signal sensor."""
-    name = "CO2 intensity"
-    if extra_name := get_extra_name(hass, entry.data):
-        name += f" - {extra_name}"
-
-    async_add_entities(
-        [
-            CO2Sensor(
-                name,
-                entry.data,
-                entry_id=entry.entry_id,
-            )
-        ],
-        True,
-    )
+    coordinator: CO2SignalCoordinator = hass.data[DOMAIN][entry.entry_id]
+    async_add_entities(CO2Sensor(coordinator, description) for description in SENSORS)
 
 
-class CO2Sensor(SensorEntity):
+class CO2Sensor(update_coordinator.CoordinatorEntity[CO2SignalResponse], SensorEntity):
     """Implementation of the CO2Signal sensor."""
 
+    _attr_state_class = STATE_CLASS_MEASUREMENT
     _attr_icon = "mdi:molecule-co2"
-    _attr_unit_of_measurement = CO2_INTENSITY_UNIT
 
-    def __init__(self, name, config, entry_id):
+    def __init__(
+        self, coordinator: CO2SignalCoordinator, description: CO2SensorEntityDescription
+    ) -> None:
         """Initialize the sensor."""
-        self._config = config
+        super().__init__(coordinator)
+        self._description = description
+
+        name = description.name
+        if extra_name := coordinator.get_extra_name():
+            name = f"{extra_name} - {name}"
+
         self._attr_name = name
         self._attr_extra_state_attributes = {ATTR_ATTRIBUTION: ATTRIBUTION}
         self._attr_device_info = {
-            ATTR_IDENTIFIERS: {(DOMAIN, entry_id)},
+            ATTR_IDENTIFIERS: {(DOMAIN, coordinator.entry_id)},
             ATTR_NAME: "CO2 signal",
             ATTR_MANUFACTURER: "Tmrow.com",
             "entry_type": "service",
         }
-        self._attr_unique_id = f"{entry_id}_co2intensity"
-
-    def update(self):
-        """Get the latest data and updates the states."""
-        _LOGGER.debug("Update data for %s", self.name)
-
-        if CONF_COUNTRY_CODE in self._config:
-            kwargs = {"country_code": self._config[CONF_COUNTRY_CODE]}
-        elif CONF_LATITUDE in self._config:
-            kwargs = {
-                "latitude": self._config[CONF_LATITUDE],
-                "longitude": self._config[CONF_LONGITUDE],
-            }
-        else:
-            kwargs = {
-                "latitude": self.hass.config.latitude,
-                "longitude": self.hass.config.longitude,
-            }
-
-        self._attr_state = round(
-            CO2Signal.get_latest_carbon_intensity(self._config[CONF_API_KEY], **kwargs),
-            2,
+        self._attr_unique_id = (
+            f"{coordinator.entry_id}_{description.unique_id or description.key}"
         )
+
+    @property
+    def state(self) -> StateType:
+        """Return sensor state."""
+        return round(self.coordinator.data["data"][self._description.key], 2)  # type: ignore
```

Comment:
What type error are we ignoring? Please specify it, eg `ignore[bla]` to make it easier to resolve in the future.